geode-connectors/src/main/java/org/apache/geode/connectors/util/internal/DescribeMappingResult.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.connectors.util.internal;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.geode.DataSerializable;
+import org.apache.geode.DataSerializer;
+
+public class DescribeMappingResult implements DataSerializable {
+  private Map<String, String> attributeMap;
+
+  public DescribeMappingResult() {
+    this.attributeMap = new HashMap<>();
+  }
+
+  public DescribeMappingResult(Map<String, String> attributeMap) {
+    this.attributeMap = attributeMap;
+  }
+
+  public Map<String, String> getAttributeMap() {
+    return this.attributeMap;
+  }
+
+  @Override
+  public void toData(DataOutput out) throws IOException {
+    DataSerializer.writeInteger(attributeMap.size(), out);
+    for (Map.Entry<String, String> entry : attributeMap.entrySet()) {
+      DataSerializer.writeString(entry.getKey(), out);
+      DataSerializer.writeString(entry.getValue(), out);
+    }
+  }
+
+  @Override
+  public void fromData(DataInput in) throws IOException {
+    int dataSize = in.readInt();
```

Comment:
instead of using in.readInt keep your code "balanced" by using DataSerializer.readInteger